sources/packages/api/src/route-controllers/application/application.system.controller.ts

```diff
@@ -295,4 +302,49 @@ export class ApplicationSystemController {
       }
     }
   }
+
+  /**
+   * Creates a CRA Income Verification record that will be waiting
+   * to be send to CRA and receive a response.
+   * @param payload information needed to create the CRA Income Verification record.
+   * @returns the id of the new CRA Verification record created.
+   */
+  @Post(":applicationId/income-verification")
+  async createIncomeVerification(
+    @Param("applicationId") applicationId: number,
+    @Body() payload: CreateIncomeVerificationDto,
+  ): Promise<number> {
+    const incomeVerification =
+      await this.incomeVerificationService.createIncomeVerification(
```

Comment:
Please add a try catch here, with userdefined exception, as the CreateIncomeVerificationDto has taxyear validation

export class CreateIncomeVerificationDto {
  @IsInt()
  @Min(2000)
  taxYear: number;
  @IsInt()
  @Min(0)
  reportedIncome: number;
}

Reply:
I believe that there is a misunderstanding about how the class validator works.
The validators will be evaluated before the code execution reaches the controller, if they fail, the API will return a nice BAD REQUEST exception.
If the code on line 317 is reached, is because all the validators passed. Does it make sense?